e2e/helm_charts_test.go

```diff
@@ -311,6 +311,53 @@ func (ts *HelmChartTestSuite) resourceWillManageAdditionalPodsForMetricsets(reso
 	return nil
 }
 
+func (ts *HelmChartTestSuite) strategyCanBeUsedDuringUpdates(strategy string) error {
+	output, err := kubectl.Run("get", "ds/"+ts.Name+"-"+ts.Name, "-o", `go-template='{{.spec.updateStrategy.type}}'`)
+	if err != nil {
+		return err
+	}
+
+	output = strings.ReplaceAll(output, "'", "")
+	if output != strategy {
+		return fmt.Errorf("There is no %s strategy to be used on updates. Actual: %s", strategy, output)
+	}
+
+	log.WithFields(log.Fields{
+		"strategy": strategy,
+		"chart":    ts.Name,
+	}).Debug("The strategy can be used during updates")
+
+	return nil
+}
+
+func (ts *HelmChartTestSuite) strategyCanBeUsedForResourceDuringUpdates(strategy string, resource string) error {
+	lowerResource := strings.ToLower(resource)
+	strategyKey := "strategy"
+	name := ts.Name + "-" + ts.Name + "-metrics"
+
+	if lowerResource == "daemonset" {
+		strategyKey = "updateStrategy"
+		name = ts.Name + "-" + ts.Name
+	}
+
+	output, err := kubectl.Run("get", lowerResource, name, "-o", `go-template='{{.spec.`+strategyKey+`.type}}'`)
```

Comment:
@kuisathaverat same here

Reply:
both are the same command, the only difference is the go-template

Reply:
I'm calling one method from the other one.